kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/ContentHostingComparator.java

```diff
@@ -94,6 +94,10 @@ public ContentHostingComparator(String property, boolean ascending, boolean is_s
 	 */
 	public int compare(Object o1, Object o2)
 	{
+		if (o1 == null && o2 == null) { return 0; }
```

Comment:
It's just that before hand we might (if sorting by rank) have blown up with a NPE if null was passed to this comparator and I don't think it really makes sense to sort null items in this sense so I'd be inline to just drop the lines. The reason being that otherwise in a years time someone else looking at the code sees that the comparator deals with nulls and so then thinks this is the correct and acceptable use.

Unless someone can think why you would want to sort a a list with a null in it?

Throwing a NPE is allowed by the Comparator contract as well.

This isn't a biggie at all and it's probably fine to go in.